cmd/thanos/sidecar.go

```diff
@@ -362,7 +375,12 @@ func runSidecar(
 			s := shipper.New(logger, reg, conf.tsdb.path, bkt, m.Labels, metadata.SidecarSource,
 				uploadCompactedFunc, conf.shipper.allowOutOfOrderUpload, metadata.HashFunc(conf.shipper.hashFunc), conf.shipper.metaFileName)
 
-			return runutil.Repeat(30*time.Second, ctx.Done(), func() error {
+			return runutil.RepeatWithJitter(30*time.Second, ctx, 0.05, func() error {
+				// Generate random delay using upload jitter.
+				jitter := DurationWithJitter(conf.shipper.uploadJitter, 0.05)
+				if jitter > 0 {
+					time.Sleep(jitter)
+				}
```

Comment:
Why we need to add another jitter here? Isn't `RepeatWithJitter` enough

Reply:
should i also remove` DurationWithJitter` function or keep it for future use ?

Reply:
Let's remove that